mysql/client/server.sls

```diff
@@ -0,0 +1,83 @@
+{%- from "mysql/map.jinja" import client with context %}
+{%- if client.enabled %}
+
+{%- for server_name, server in client.get('server', {}).iteritems() %}
+
+{%- set connection_args = {'host': server.admin.host,
+                           'port': server.admin.get('port', '3306'),
+                           'user': server.admin.user,
+                           'pass': server.admin.get('password', ''),
+                           'charset': server.admin.get('encoding', 'utf8'),
+                           'db': server.admin.get('database', 'mysql')
+} %}
+
+{%- for database_name, database in server.get('database', {}).iteritems() %}
+
+mysql_{{ server_name }}_database_{{ database_name }}:
+  mysql_database.present:
+  - name: {{ database_name }}
+  - character_set: {{ database.get('encoding', 'utf8') }}
```

Comment:
What about using this instead of specifying each connection argument separately?

```
{%- for key, value in connection_args.iteritems() %}
- connection_{{ key }}: {{ value }}
{%- endfor %}
```

Maybe make it a macro to simplify even more.